### DISCUSSION TITLE
How to inherit section number into enumerated items?

### DISCUSSION THREAD

**Discussion by ckunte** [typst/typst#4844] [Q&A]
I am writing a specification that has unique numbering requirement. I would like numbered (enumerated) items under the heading to inherit the section number prefix, for example:

```
1. Introduction

R1.1 The following words in the basis of design shall mean ...

2. Field Layout

R2.1 All geographical positions shall be referenced to the datum ...

R2.2 All projected data for the project shall be projected to zone x.
```
Currently I am using this, but when a new section is introduced, the numbering resets, which I would like to avoid.

```typ
#set enum(numbering: "R1.1", full: false)
```
Would be grateful for advise on how to do this in Typst.

**Comment by mattj23** [comment]
I know this is old but I had a similar use case and I was able to figure it out.

This is a little bit simplified because I only cared about the numbers themselves and wanted to join the heading numbers as a prefix to the `enum` full numbers.  For anyone who finds this question later, you should be able to adapt this to get what you want.

The trick has two parts:

1. Use a function for the `numbering` argument of the `enum` element function.  When you pass it a function instead of the formatting string, it will call that function with a variable number of arguments in which each argument is an integer from the numbering of that `enum.item`.  
2. Define your own function to give to `numbering` that grabs the outer `context` and uses the heading counter to get the numbering of the current section.  Then join the heading numbering and the `enum.item` numbering into one big array of integers and convert them to a string however you please.

```typst
#let heading_prefix_numbering(..args) = context {
  let chain = counter(heading).get() + args.pos()
  return chain.map(str).join(".")
}

#set enum(full: true, numbering: heading_prefix_numbering)
```

- So from `counter(heading).get()` you will get something like `(1, 2, 3)`

- The framework will have called something that looks like `heading_prefix_numbering(4, 5, 6)`, so `args.pos()` would be `(4, 5, 6)`

- By adding them together you'll get `(1, 2, 3, 4, 5, 6)`

- Convert them to strings with `.map(str)` and then `.join(".")` to get `"1.2.3.4.5.6"`
